framework/src/kernels/EigenKernel.C

```diff
@@ -36,11 +36,6 @@ EigenKernel::EigenKernel(const InputParameters & parameters) :
     _eigen_sys(dynamic_cast<EigenSystem *>(&_fe_problem.getNonlinearSystem())),
     _eigenvalue(NULL)
 {
-}
-
-void
-EigenKernel::initialSetup()
```

Comment:
Moving all of this to the constructor is OK but means that we are definitely requiring that the Executioner exists before this object is even built!